ignite/distributed/utils.py

```diff
@@ -0,0 +1,316 @@
+from typing import Optional, Union, Tuple
+import socket
+from functools import wraps
+
+import torch
+import torch.distributed as dist
+
+from ignite.utils import setup_logger
+from ignite.distributed.comp_models import registered_computation_models, _SerialModel, has_xla_support
+
+
+_model = _SerialModel()
+
+
+def _sync_model_wrapper(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if isinstance(_model, _SerialModel):
+            _sync_model()
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def _sync_model():
+    global _model
+    # TODO: to implement
+    pass
+
+
+@_sync_model_wrapper
+def device() -> Union[torch.device, str]:
+    """Returns current device according to current distributed configuration.
+
+    - `cpu` if no distributed configuration or native gloo distributed configuration
+    - `cuda:local_rank` if native nccl distributed configuration
+    - `xla` device if XLA distributed configuration
+
+    Returns:
+        torch.device or str
+    """
+    return _model.device()
+
+
+@_sync_model_wrapper
+def backend() -> Optional[str]:
+    """Returns computation model's backend.
+
+    - `None` for no distributed configuration
+    - "nccl" or "gloo" or "mpi" for native torch distributed configuration
+    - "xla-tpu" for XLA distributed configuration
+
+    Returns:
+        str or None
+    """
+    return _model.backend()
+
+
+def available_backends() -> Tuple[str]:
```

Comment:
Here, "available" maybe be ambiguous : return a list of backends which could not be "available" in a sense of `dist`  (is_mpi_available for instance) or `xla`. Maybe it would be better to filter the list and return effectively
available backends ?